docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+      is required for objects w/o finalizers that need to be deleted
+      from the managed clusters (WECs)
+- Starting & stopping informers dynamically based on creation or
+  deletion of CRDs (which add/remove APIs on the WDS).
+- One client connected to the WDS space and one (or more in the future)
+  to connect to one or more OCM shards.
+  - The WDS-connected client is used to start the dynamic
+    informers/listers for most API resources in the WDS
+  - The OCM-connected client is used to start informers/listers for OCM
+    ManagedClusters and to copy/update/remove the wrapped objects
+    into/from the OCM mailbox namespaces.
+
+Currently there are two controllers in the KubeStellar controller
+manager: the placement controller and the status controllor.
+
+### Placement Controller
+
+The Placement controller is responsible for watching workload and 
+placement objects, and wrapping and delivering objects to the ITS
+based on placement policies.
+
+The architecture and the event flow for create/update object events is
+illustrated in Figure 3 (some details might be omitted to make the flow easier
+to understand).
+
+<figure>
+  <img src="assets/placement-controller.png"  alt="Placement Controller">
+  <figcaption align="center">Figure 3 - Placement Controller</figcaption>
+</figure>
+
+At startup, the controller code sets up the dynamic informers, the event
+handler and the work queue as follows:
+
+- lists all API resources
+- Filters out some resources
+- For each resource:
+  - Creates GVK key
+  - Registers Event Handler
+  - Starts Informer
+  - Indexes informer and lister in a map by GVK key
+- Waits for all caches to sync
+- Starts N workers to process work queue
+
+The reflector is started as part of the informer and watches specific
+resources on the WDS API Server; on create/update/delete object events it
+puts a copy of the object into the local cache. The informer invokes the
+event handler. The handler implements the event handling functions
+(AddFunc, UpdateFunc, DeleteFunc)
+
+A typical event flow for a create/update object event will run as
+follows:
+
+1.  Informer invokes the event handler AddFunc or UpdateFunc
+
+2.  The event handler does some filtering (for example, to ignore update
+    events where the object content is not modified) and then creates a
+    key to store a reference to the object in the work queue. The key
+    contains the GVK key used to retrieve a reference to the informer
+    and lister for that kind of object, and a namespace/name key to
+    retrieve the actual object. Storing the key in the work queue is a
+    common pattern in client-go as the object may have changed in the
+    cache (which is kept updated by the reflector) by the time a worker
+    gets a copy from the work queue. Workers should always receive the
+    key and use it to retrieve the object from the cache.
+
+3.  A worker pulls a key from the work queue, and then does the
```

Comment:
What about a workload object update that shrinks the set of Placements that match?

There is special handling for `Namespace` objects, right?

What about cluster add/change/delete?

Reply:
> What about a workload object update that shrinks the set of Placements that match?
This is not currently not covered. Should be listed in "not currently addressed". I suggest we open a new issue to discuss approaches for this.

> There is special handling for Namespace objects, right?
Can you be more specific? What is the expectation for a namespace object ? At this time it is treated like any other object.

> What about cluster add/change/delete?
This is not currently not covered. Should be listed in "not currently addressed". I suggest we open a new issue to discuss approaches for this.

Reply:
I created #1674 for this.